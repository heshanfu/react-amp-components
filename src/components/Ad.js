import React, { Component } from 'react'
import { string, oneOfType, arrayOf, node } from 'prop-types'
import { Helmet } from 'react-helmet'

import pick from '../utils/pick'
import commonPropTypes from './commonPropTypes'

class Ad extends Component {
  static propTypes = {
    ...commonPropTypes,
    type: string.isRequired,
    'data-slot': string.isRequired,
    children: oneOfType([arrayOf(node), node])
  };

  static defaultProps = {
    children: null
  };

  get properties() {
    return pick(this.props, ...Object.keys(commonPropTypes))
  }

  render() {
    const { children } = this.props
    return [
      <Helmet key="helmet">
        <script
          async=""
          custom-element="amp-ad"
          src="https://cdn.ampproject.org/v0/amp-ad-0.1.js"
        />
      </Helmet>,
      <amp-ad {...{ ...this.properties }}>{children}</amp-ad>
    ]
  }
}

export default Ad
