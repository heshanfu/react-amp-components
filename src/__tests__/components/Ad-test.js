import React from "react";
import Ad from "../../components/Ad";
import Helmet from "../../utils/Helmet";
import { renderComponent } from "../test-utils";

describe("Bind", () => {
  it("works", () => {
    const res = renderComponent(
      <Ad data-slot="xxxx" width="1150" height="160" type="doubleclick" />
    );
    expect(res.toJSON()).toMatchSnapshot();
  });

  it("injects the right script tag", () => {
    renderComponent(
      <Ad data-slot="xxxx" width="1150" height="160" type="doubleclick" />
    );
    Helmet.canUseDOM = false;
    const staticHead = Helmet.renderStatic();
    expect(staticHead.scriptTags).toMatchSnapshot();
  });
});
