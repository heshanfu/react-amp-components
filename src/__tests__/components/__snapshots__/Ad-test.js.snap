// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bind injects the right script tag 1`] = `
Array [
  Object {
    "async": "",
    "custom-element": "amp-ad",
    "src": "https://cdn.ampproject.org/v0/amp-ad-0.1.js",
  },
]
`;

exports[`Bind works 1`] = `
<amp-ad
  data-slot="xxxx"
  height="160"
  type="doubleclick"
  width="1150"
/>
`;
